examples/fidelity.py

```diff
@@ -0,0 +1,82 @@
+"""Simulates the fidelity estimation.
+
+Direct Fidelity Estimation from Few Pauli Measurements
+https://arxiv.org/abs/1104.4695
+
+Practical characterization of quantum devices without tomography
+https://arxiv.org/abs/1104.3835
+"""
+
+import cirq
+import heapq
+import itertools
+import math
+import numpy
+
+
+def build_circuit():
+  qubits = cirq.LineQubit.range(3)
+  circuit = cirq.Circuit(
+      cirq.Z(qubits[0])**0.25,  # T-Gate, non Clifford.
+      cirq.X(qubits[1])**0.123,
+      cirq.X(qubits[2])**0.456)
+  return circuit, qubits
+
+
+def simulate_trace(circuit, pauli_gates, qubits):
+  simulator = cirq.DensityMatrixSimulator()
+
+  n = len(pauli_gates)
+  d = 2**n
+
+  trace = 0
+  for x in range(d):
+    xbin = numpy.binary_repr(x, width=n)
+
+    rotated_initial_state = cirq.final_wavefunction(
+        cirq.DensePauliString(pauli_gates),
+        qubit_order=qubits,
+        initial_state=x)
+
+    y = simulator.simulate(circuit, initial_state=rotated_initial_state).measurements['y']
+    trace += sum([int(xbin[i]) == y[i] for i in range(n)])
+
+  return trace
+
+
+def main():
+  circuit, qubits = build_circuit()
+  noisy_circuit = circuit.with_noise(cirq.amplitude_damp(0.01))
```

Comment:
Move this down a line so that the measurement appending is not done twice.

Reply:
I think it's moot now, since I followed your advice (I think) below to set the noise in the simulator.